src/lazy.js

```diff
@@ -31,14 +32,24 @@ const lazy = (options = {}) => {
 
     const onIntersection = entries => {
         entries.forEach(entry => {
-            if (entry.intersectionRatio > 0) {
-                const el = entry.target;
+            const el = entry.target;
+            if (entry.isIntersecting) {
                 load(el);
+            } else {
+                onNonIntersection(el);
             }
         });
     };
 
-    const observe = els => {
+    const onNonIntersection = el => {
+        if (settings.onNonIntersectionCallback !== null && 
+            initialIntersection === false) {
```

Comment:
What happens if the element is within the viewport at page load, and then the user scrolls down the page and the element leaves the viewport? You'd probably want the non-intersection callback to fire in that scenario, but I don't think it would because `initialIntersection` would be true.

Reply:
Maybe also setting initialIntersection to false in your load() method would solve this problem?